decidim-core/app/models/decidim/share_token.rb

```diff
@@ -10,7 +10,10 @@ class ShareToken < ApplicationRecord
     belongs_to :user, foreign_key: "decidim_user_id", class_name: "Decidim::User"
     belongs_to :token_for, foreign_type: "token_for_type", polymorphic: true
 
-    after_initialize :generate, :set_default_expiration
+    after_initialize :generate
+    before_create :set_default_expiration, unless: :expiration_blank
+
+    attr_accessor :expiration_blank
```

Comment:
Having tokens without expiration raises a security leak. We're leaving the responsibility to regenerate the token to the memory of the process administrator. I don't think is a good idea.

Reply:
You're right @tramuntanal, we should limit this @entantoencuanto

Reply:
@entantoencuanto this doesn't seem to be solved yet!

Reply:
Is this really a problem? In our use case enforcing an expiration date would break the desired behaviour. Or we'd have to enter a date many years from now, which is the equivalent of not having an expiration date...

Reply:
@andreslucena could you chime in here please?